polyglot.js

```diff
@@ -71,29 +104,58 @@ class PolyGlot {
 			actors.push(game.user.character);
 		for (let actor of actors) {
 			try {
-				// Don't duplicate the value in case it's a not an array
-				for (let lang of actor.data.data.traits.languages.value)
-				    this.known_languages.add(lang)
-			} catch (err) { 
+				switch(game.system.id.toLowerCase()){
+					default:
+						// Don't duplicate the value in case it's a not an array
+						for (let lang of actor.data.data.traits.languages.value)
+								this.known_languages.add(lang)
+						break;
+					case "wfrp4e":
+						let re = /Language \((.+)\)/i;
+						for (let item of actor.data.items){
+							let found = item.name.match(re);
+							if(found) {
+								// adding only the descriptive language name, not "Language (XYZ)"
+								this.known_languages.add(found[1]);
+							}
+						}
```

Comment:
Other than the usual coding style changes I mentioned previously, I think this would benefit from using array functions :
```js
const user_languages = actor.data.items.map(i => {
    const match = item.name.match(/Language \((.+)\)/i);
    if (match) return match[1];
}).filter(i => !!i);
this.known_languages= new Set([...this.known_languages, ...user_languages]);
```
I'm not a fan of recreating the Set every time, so maybe the way you did it is better after all. But using a similar map/filter solution would be nice in the `getLanguages` function when you redo it.